airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -350,7 +351,12 @@ def _dynamic_stream_configs(
                 if "type" not in dynamic_stream:
                     dynamic_stream["type"] = "DeclarativeStream"
 
-                dynamic_stream_configs.append(dynamic_stream)
+                # Ensure that each stream is created with a unique name
```

Comment:
Why could we have duplicates here? It feels like we would need more context if we keep those lines. I see from the test that an API could return twice the same information. Is this a case we've seen with a source? With my limited understanding of the problem, this is more like a config error than something we should handle by silently removing some streams.

Reply:
I agree. This can help identify those issues as soon as possible. 
Updated